### PR TITLE
Allow gardenlet to delete ShootStates in SeedAuthorizer

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area security
/kind bug
/merge squash

**What this PR does / why we need it**:
After #4100, the gardenlet must be allowed to delete `ShootState` resources.
Additionally, in order to reduce complexity, the `ShootState` event handler has been removed from the resource dependency graph while the `ShootState` vertex is now created in the `Shoot` event handler. This is to always allow a gardenlet to delete `ShootState`s (even if they were already gone before).

**Which issue(s) this PR fixes**:
Part of #1723

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/admissioncontroller/webhooks/auth/seed/graph/eventhandler_shootstate.go [hunk 1]
```diff
@@ -1,74 +0,0 @@
...
-	"context"
-	"time"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	toolscache "k8s.io/client-go/tools/cache"
-	"sigs.k8s.io/controller-runtime/pkg/cache"
-)
-
-func (g *graph) setupShootStateWatch(_ context.Context, informer cache.Informer) {
-	informer.AddEventHandler(toolscache.ResourceEventHandlerFuncs{
```
Comment (ialidzhikov) [inline pkg/admissioncontroller/webhooks/auth/seed/graph/eventhandler_shootstate.go:27]: Does this mean that the admission-controller does no longer need to `get`, `list` and `watch` permissions for `shootstates`?

Comment (rfranzke) [reply pkg/admissioncontroller/webhooks/auth/seed/graph/eventhandler_shootstate.go:27]: I think yes, nice catch, I'll take care.

**Review by ialidzhikov — COMMENTED** [review]
/lgtm
/needs second-opinion

**Review by ialidzhikov — APPROVED** [review]

**Review by timuthy — APPROVED** [review]
/lgtm

**Pull request merged**